packages/roosterjs-editor-types/lib/interface/TableSelection.ts

```diff
@@ -13,4 +13,9 @@ export default interface TableSelection {
      * Last Cell selection x and y position in the table
      */
     lastCell: Coordinates;
+
+    /**
+     * Path of selection done by the user.
+     */
+    selectionPath: Coordinates[];
```

Comment:
We already have firstCell/lastCell, why we still need selection path?

Reply:
firstCell and lastCell are not always the actual cell selected in the table since we normalize the x,y. This is the case when selection starts from top right and finishes in the left bottom, like this:

![image](https://user-images.githubusercontent.com/8291124/188987766-02c09b2f-1ead-4fa5-8cfb-86591a413f39.png)

Normalized selection would be:
firstCell: 1,1
lastCell: 8,7.

But the selection was from
firstCell: 8,1
lastCell: 1,7

I was having some issues with that, so I added the selection Path so we can get the first and last cell that the user selected.  

Any feedback for a better implementation?

Reply:
Can you help my understand why you need to know the first/last cell user selected?

Reply:
For some scenarios like this:
> For the second case, I tried in Word desktop, and it allows me to select a non-rectangle selection:
> 
> <img alt="image" width="468" src="https://user-images.githubusercontent.com/23065085/188733031-fe8df2e4-a851-4117-92d0-ccdd3f139713.png">
> 
> And I think Word's behavior is better, otherwise I have no way to only select cell 0 and 1 with out 2 and 3.

Word behavior depends on where the selection started as mentioned here: 

 
> For example: If selection end is a merged cell does this: ![MergeSel1](https://user-images.githubusercontent.com/8291124/188902343-5e225e5f-09f4-49c5-a748-f080e320522f.gif) 
> 
> If selection starts from merged: ![MergeSel2](https://user-images.githubusercontent.com/8291124/188902441-f170d1f9-e30f-4677-97d7-b296f9742852.gif)